django_pg_simple_hll/migrations/0002_custom_hashing.sql

```diff
@@ -0,0 +1,106 @@
+-- drop some methods that we will overwrite
+DROP AGGREGATE IF EXISTS hll_cardinality(anyelement, int);
+DROP AGGREGATE IF EXISTS hll_cardinality(anyelement);
+DROP FUNCTION IF EXISTS hll_bucket(int [], anyelement);
+DROP FUNCTION IF EXISTS hll_bucket(int [], anyelement, int);
+
+
+-- Default hashing function, it turns any input into an unsigned 32 bit integer
+CREATE OR REPLACE FUNCTION hll_hash(input anyelement) RETURNS int
+LANGUAGE sql IMMUTABLE STRICT PARALLEL SAFE AS $$
+SELECT HASHTEXT(input::text) & ~(1 << 31)
```

Comment:
Yep, but in the case where you have bit `x` that looks like 1000...0000 (1 followed by however many zeros) then `x - 1` and `~x` are the same

Reply:
ah, i see

I'd have to test this. But there is a couple of things that make me think that this is going to be tricky:
- we want to subtract 1 from the largest representable integer in the data type, so I suspect we'd have to use a different data type and then cast it.
- this is the signing bit that we are trying to unset

So, I'm tempted to leave this as is because I know it works

Reply:
kl, yeah doesn't sound worth the effort